arraycontext/impl/pyopencl/taggable_cl_array.py

```diff
@@ -0,0 +1,112 @@
+"""
+.. autoclass:: TaggableCLArray
+.. autoclass:: Axis
+
+.. autofunction:: to_tagged_cl_array
+"""
+
+import pyopencl.array as cla
+from typing import FrozenSet, Union, Sequence, Optional, Tuple
+from pytools.tag import Taggable, Tag, tag_dataclass
+
+
+@tag_dataclass
```

Comment:
I'm not sure this is itself a tag, so I don't think the `@tag_dataclass` belongs.

Reply:
Oops, yep. Amended.